### ISSUE TITLE
IJCH problems with API

### ISSUE THREAD

**Issue by ClaraDOAJ** [DOAJ/doaj#1313]
IJCH is currently running on PJS 3.0.2 and there is an inbuilt plugin of DOAJ with API. I generated a new key for them but they are still having problems.

See error:

Deposit was not successful! The DOAJ API returned an error: '1 - {"status": "unauthorised", "error": "An API Key is required to access this. (ref: 6885413c-3fc1-11e7-aa44-04015bfb3501)"}'.
 
Notification
Deposit was not successful! The DOAJ API returned an error: '1 - 
301 Moved Permanently

nginx/1.4.6 (Ubuntu)

**Comment by richard-jones** [comment]
@ClaraDOAJ Do you know what operation they were trying to do when this happened?

**Comment by richard-jones** [comment]
Here's a few things to check to make sure that they should be able to access the API:

* That they have an API key set in their user account (which you have done), and that they are using the right API key to access the API
* That they have the "api" role set against their user account
* That their system is using the right field when sending the api key: "api_key"

Can you give us the account name, and I'll check their user account setup.

**Comment by ClaraDOAJ** [comment]
09717587

**Comment by ClaraDOAJ** [comment]
More detail;:
![image](https://cloud.githubusercontent.com/assets/22593616/26732379/b331f4f2-47b7-11e7-807f-7dfec87633c0.png)


![image](https://cloud.githubusercontent.com/assets/22593616/26732355/a24d32aa-47b7-11e7-9045-4c91158ee6f2.png)

![image](https://cloud.githubusercontent.com/assets/22593616/26732367/ad39c494-47b7-11e7-8e4a-781f8bbeef24.png)

**Comment by emanuil-tolev** [comment]
I think some of the developers behind a technical library used by the OJS platform have contacted me via github. The original error at the top of this issue might not be relevant anymore - we shall see. I'm just recording it here since it looks like it might be relevant to this issue.

This is the thread (it is very technical): https://github.com/pkp/pkp-lib/issues/2543#issuecomment-304570146

They are currently (at the time of writing) getting an unexplained 502 bad gateway error despite the website being functional - it only happens for their specific API request. I need to investigate that further.

**Comment by emanuil-tolev** [comment]
A potential fix that cuts processing time in half is now rolled out to live. I need to get back to those folks. I'll look at increasing the timeout slightly as well, to make the errors they are experiencing less likely.

**Comment by emanuil-tolev** [comment]
I've got back to the PKP devs. Unfortunately despite our efforts we're still seeing occasional timeouts with the articles they wanted to submit. Whether it times out or not seems to depend on server load. If it takes more than 40 seconds, the web worker is killed. This is a better situation than 100% timing out all the time, which is where we started, but is still not good enough to recommend to the PKP plugin developers.

So, I offered a work-around: could they use the individual article submission endpoint? Simply looping around the 1000 articles they were hoping to send and sending them 1 by 1 would get around any time-out issues they had with the Bulk upload.

This option does mean higher network overheads, so I've left to them to decide whether this is acceptable.
